packages/replayio/src/utils/recordings/upload/uploadRecording.ts

```diff
@@ -130,63 +127,26 @@ export async function uploadRecording(
   recording.uploadStatus = "uploaded";
 
   if (processAfterUpload) {
-    debug("Processing recording %s ...", recording.id);
-
-    updateRecordingLog(recording, {
-      kind: RECORDING_LOG_KIND.processingStarted,
-    });
-
-    recording.processingStatus = "processing";
+    debug("Begin processing recording %s ...", recording.id);
 
     try {
       await retryWithExponentialBackoff(() => processUploadedRecording(client, recording));
-
-      updateRecordingLog(recording, {
-        kind: RECORDING_LOG_KIND.processingFinished,
-        server: replayServer,
-      });
-
-      recording.processingStatus = "processed";
     } catch (error) {
-      debug(`Processing failed for recording ${recording.id}`);
+      debug(`Failed to begin processing recording ${recording.id}`);
 
-      recording.processingStatus = "failed";
-
-      // Processing failed but the recording still uploaded successfully
+      // Processing may have failed to start, but the recording still uploaded successfully
     }
   }
 }
 
 async function processUploadedRecording(client: ProtocolClient, recording: LocalRecording) {
-  const result = await Promise.race([
-    createSession(client, {
-      recordingId: recording.id,
-    }),
-    wait(10_000),
-  ]);
-  if (result == null) {
-    throw new Error("Timed out waiting for createSession");
-  }
-
-  const { sessionId } = result;
-
   await client.waitUntilAuthenticated();
 
-  debug(`Processing recording for session ${sessionId}`);
+  debug(`Processing recording ${recording.id}`);
 
-  try {
-    await client.sendCommand<ensureProcessedParameters, ensureProcessedResult>({
-      method: "Session.ensureProcessed",
-      params: {},
-      sessionId,
-    });
-
-    debug("Processed recording %s", recording.id);
-  } finally {
-    debug("Releasing session %s", sessionId);
-
-    await releaseSession(client, { sessionId });
-  }
+  // Processing can take a while
+  // In some cases it's nicer to give users a URL sooner rather than waiting for processing to complete
+  processRecording(client, { recordingId: recording.id });
```

Comment:
I'm not sure what processing is yet in this context - does it make any sense to wait for it to be completed at any later point in time? Or can we truly treat it as fire-and-forget since it's all up to the backend?

Can this somehow fail/throw - in other words, should we attach a `.catch(noop)` to this or smth?

Reply:
> Can this somehow fail/throw - in other words, should we attach a .catch(noop) to this or smth?

We are catching (higher up) and retrying.
```js
    try {
      await retryWithExponentialBackoff(() => processUploadedRecording(client, recording));
    } catch (error) {
      debug(`Failed to begin processing recording ${recording.id}`);

      // Processing may have failed to start, but the recording still uploaded successfully
    }
```

I can probably inline this stuff now though. It was separate before because I was also creating a session (which could itself fail). I'll do a quick cleanup pass so it's clearer.